drivers/i3c/master.c

```diff
@@ -1722,6 +1722,7 @@ static int i3c_master_early_i3c_dev_add(struct i3c_master_controller *master,
 {
 	struct i3c_device_info info = {
 		.static_addr = boardinfo->static_addr,
+		.dyn_addr = boardinfo->init_dyn_addr,
```

Comment:
Billy's comment: 
The dynamic address will be assigned only after the SETDASA succeeds; until then, the device should only respond to its static address.
https://github.com/AMDESE/linux-aspeed/blob/integ_sp7/drivers/i3c/master.c#L1740-L1745

Reply:
> Billy's comment: The dynamic address will be assigned only after the SETDASA succeeds; until then, the device should only respond to its static address. https://github.com/AMDESE/linux-aspeed/blob/integ_sp7/drivers/i3c/master.c#L1740-L1745

@ojayanth  - Following check fails when hci_dat_v1_alloc_entry() is called with zero dynamic address. 
hci_dat_v1_alloc_entry
https://github.com/AMDESE/linux-aspeed/blob/integ_sp7/drivers/i3c/master/mipi-i3c-hci/dat_v1.c#L100C1-L101C18

For SBRMI/SBTSI devices, driver need to send the SETDASA command as part of the early init process during the i3c bus registration. Before sending the SETDASA command driver reserves an entry in the device address table with dynamic address as the index which fails with out my fix and subsequently SETDASA command will not be sent, and the device nodes will not be created.  Please check with Billy what could be a suggested fix. Below is the call stack:

i3c_master_register()
i3c_master_bus_init()
i3c_master_early_i3c_dev_add() 
i3c_hci_attach_i3c_dev() ==>Here instead of using the info.dyn_addr as parameter, we can take the dynamic address mentioned in the device tree. this will avoid change to the toplevel driver (master.c). 
hci_dat_v1_alloc_entry()